developer_docs/foreman_dev_setup.asciidoc

```diff
@@ -0,0 +1,115 @@
+[[Containers]]
+= Install From Source
+:toc: right
+:toclevels: 5
+
+Installing the latest development code:
+Foreman uses Bundler to install dependencies and get up and running. This is the preferred way to get Foreman if you want to benefit from the latest improvements. By using the git repository you can also upgrade more easily. You will need to have Bundler installed manually for now (check your distribution repositories, or install it via rubygems).
+
+[[prerequisites]]
+== Prerequisites
+Foreman will run with the following requirements (aside from rubygem dependencies):
+
+* Ruby v2.7.x
+* NodeJS v12 or v14
+* NPM v6
+* PostgreSQL v13
+
+It’s recommended to install https://github.com/rbenv/rbenv[rbenv] to manage ruby, and https://github.com/nvm-sh/nvm[nvm] to manage node and npm on your system.
+Before installing PostgreSQL, check first that you don't already have it installed (by running `rpm -q postgresql`).
+
+You will want to make sure that you have postgresql-devel installed so the database gems can install properly.
+Also, you would also need gcc, ruby-devel, libxml-devel, libxslt-devel and libvirt-devel packages.
+
+For RHEL7 or clones, you can issue the following commands to install all required packages:
+[source, bash]
+....
+yum groupinstall "Development Tools" "Development Libraries"
+yum -y install gcc-c++ git ruby ruby-devel rubygems \
+    libvirt-devel postgresql-devel openssl-devel \
+    libxml2-devel libxslt-devel zlib-devel \
+    readline-devel systemd-devel tar libcurl-devel
+....
+
+[[Setup]]
+== Setting the repository
+[source, bash]
+....
+# On Github create your own fork
+git clone https://github.com/{GITHUB_USER}/foreman.git -b develop
+cd foreman
+cp config/settings.yaml.example config/settings.yaml
+cp config/database.yml.example config/database.yml
+git remote add upstream git@github.com:theforeman/foreman.git
+gem install bundler
+bundle install
+npm install
+....
+
+[[Database]]
+== Setting the database
+It is important that config/database.yml is set to use the correct database in the “development” block.
+Rails (and subsequently Foreman) will use these connection settings under “development” to manage the database it uses and setup the necessary schema.
+
+If the default version of the database.yml file isn't working for you, try editing it with the following:
+[source, ruby]
+....
+development:
+ adapter: postgresql
+ host: localhost
+ username: postgres
+ password: password
+ pool: 5
+ database: foreman_development
+
+test:
+ adapter: postgresql
+ host: localhost
+ username: postgres
+ password: password
+ pool: 5
+ database: foreman_test
+....
+
+Set up database schema, precompile assets and locales:
+[source, ruby]
+....
+bundle exec rake db:create # if not already created
+bundle exec rake db:migrate
+bundle exec rake db:seed assets:precompile locale:pack webpack:compile
+....
+
+The db:seed step will print out the default admin password, record this in order to log in later.
```

Comment:
Does it still happen? Better use the `bundle exec rake permissions:reset password=MYPASS`

Reply:
Should I check if it happens before deleting it?

Reply:
I didn't forget this, but I didn't finish installing PostgreSQL on my VM, so I will update it when I am done.